ecosystem/scenarios/utils.py

```
@@ -0,0 +1,260 @@
+
+    # Create a minimal setuptools mock to capture requirements
+    requirements = {"install_requires": [], "extras_require": {}}
+
+    def mock_setup(**kwargs):
+        """Mock setup function to capture requirements."""
+        if "install_requires" in kwargs:
+            requirements["install_requires"] = kwargs["install_requires"]
+        if "extras_require" in kwargs:
+            requirements["extras_require"] = kwargs["extras_require"]
+
+    def mock_find_packages(**kwargs):
```

Pylint (3.3.7)

[warning] Unused argument 'kwargs'

(W0613)